### PR TITLE
Ids 954 respond to get requests

### PR DESCRIPTION
**Pull request by LibbyLi667**
This PR adds new public models for projects, research services and members to include essential nested data for projects.

### CODE REVIEW

File: src/models/project.py [hunk 1]
```diff
@@ -66,3 +75,12 @@ class Project(BaseProject, table=True):
         back_populates="project",
         cascade_delete=True,
     )
+
+
+class ProjectWithDriveMember(BaseProject):
+    """Public model for project with drive and member information."""
+
+    id: int
+    codes: list[Code]
```
Comment (uoa-noel) [inline src/models/project.py:84]: CodePublic here could be useful, if CodePublic excludes the id field then we don't have to return the unnecessary id.

File: src/models/services.py [hunk 2]
```diff
@@ -48,3 +49,17 @@ class InputServices(SQLModel):
...
+
+    name: str
+    allocated_gb: float
+    free_gb: float
+    used_gb: float
+    percentage_used: float
+    date: datetime
+    first_day: datetime
+    last_day: Optional[datetime]
+    manifest: Manifest
```
Comment (uoa-noel) [inline src/models/services.py:65]: One other way to do could be to extract the common fields between ResearchDriveServicePublic and ResearchDriveService into a separate class, then inherit from it for both classes. See BaseProject, InputProject and Project in project.py for an example. But this is OK as is!

**Review by uoa-noel — CHANGES_REQUESTED** [review]
Just a small change around the CodePublic class, otherwise looks good!

File: src/models/project.py (hunk 1 above)
Comment (LibbyLi667) [reply src/models/project.py:76]: Yep! Just removed it as it's not necessary

File: src/models/services.py (hunk 2 above)
Comment (LibbyLi667) [reply src/models/services.py]: Good idea. Just added a BaseDriveService as the base class.

**Review by uoa-noel — APPROVED** [review]
LGTM!

**Pull request merged**